### ISSUE TITLE
Remove `twisted.python.logger.LegacyLogger`.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7639]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#7639|
|Type|enhancement|
|Created|2014-09-15 23:10:46Z|
|Branch|https://github.com/twisted/twisted/tree/log-remove-legacylogger-7639|

exarkun asked in [#7596](https://github.com/twisted/twisted/issues/7596) why there is a `twisted.python.logger.LegacyLogger` when there is still `twisted.python.log.LogPublisher` around, and on reflection I think it probably shouldn't be there unless and until someone asks for it, and maybe still not then.

The old API is still there, it forwards to the new framework, and it behaves in it's way and gets a lot of testing by virtue of being the thing presently used by everyone.  `twisted.python.logger.LegacyLogger` isn't used anywhere yet and may not be necessary anyway, so let's pull it (and the references to it in the docs).  No point in having extra public API here, and we can always dig it back out later.

Tagging for 14.1 so we clear this up before its first release.

<details><summary>Searchable metadata</summary>

```
trac-id__7639 7639
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__normal normal
milestone__New_Logging_System New Logging System
branch__branches_log_remove_legacylogger_7639 branches/log-remove-legacylogger-7639
branch_author__ 
status__closed closed
resolution__fixed fixed
component__logger logger
keywords__ 
time__1410822646650307 1410822646650307
changetime__1481165792710259 1481165792710259
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez set status to `assigned`|
|-|-|

Waiting for [#7596](https://github.com/twisted/twisted/issues/7596) to merge before I fiddle with this.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez commented|
|-|-|

Derp.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

In my mind, the purpose of `LegacyLogger` was to reduce the amount of code changes you need to make in order to start receiving some of the benefits of the new logging system.  Particularly the presence of the namespace key.

There's also the fact that it enables a gradual migration where you can instantiate a shiny new `Logger` at the top of your module, configure it appropriately, and then wrap a `LegacyLogger` around it so that the old-style log messages you haven't updated yet are still carrying the information associated with that underlying new-style `Logger` instance.

A `LegacyLogger` also has less global state; the presence of the old `log` function accomplishes compatibility by having a whole big initialized old-style logging _system_ whereas a `LegacyLogger` is a fairly isolated logging _object_.

That said, I don't know if `LegacyLogger` does this job super well.  Maybe the namespace key is the only valuable contribution?  I like it, but perhaps the right way to motivate me to get it to do its job well is to delete it immediately, and I can try opening a new ticket to land an enhanced version of it :-).  (Particularly a version of it which allows for customization of the log-level that it's logging at, "info" might not be appropriate very often given Twisted's reluctance to log very much…)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

Yeah, I guess I don't see the benefits as being all that, and I'm concerned that it hasn't gotten enough usage to be bug-free and/or the right functionality, so I'm hesitant to commit to it as supported API.

The gradual migration was something I valued before the old logging forwarded to the new logging (in CalendarServer, new logging forwarded to old logging because we wanted to keep Twisted's functionality in place without changing Twisted).  I don't find it compelling as a half-ported-file option.  If you want to switch to the new logging module, I think it's un-sane to only fix part of it, because it could be really confusing to have both styles in play in the same file, and it's not that hard to clean up even a pretty long file.

I agree about the less global state, but I do think the best way to get there is just to adopt `Logger`.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Replying to [wsanchez](#note_4):

> I agree about the less global state, but I do think the best way to get there is just to adopt `Logger`.

OK, yes.  To the extent that I have time to spend time on logging before 14.1, it does seem like it would be more worthwhile to spend time porting, say, the reactor core, or twisted mail, or something, to do logging with `Logger` directly than trying to come up with some souped-up `LegacyLogger`.

I'm not _super_ happy about deleting it but I guess I'm on board.  Less public API is better public API.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_7"></a>|@wsanchez commented|
|-|-|

r43438 removes `LegacyLogger`.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @glyph<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez commented|
|-|-|

Buildbot is green other than pre-existing pydoctor/twistedchecker noise on [branches/log-booyah-6750-5.](branches/log-booyah-6750-5.)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_11"></a>|@glyph removed owner|
|-|-|

Un-assigning all review tickets assigned to me.  Unless there's a specific reason _I_ need to review something, let's leave them unassigned :-).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban commented|
|-|-|

Hm... is the branch ok? 

On github it shows 61 changed files  with 7,487 additions and 541 deletions. ... those huge additions does not look like they are in sync with ticket subject.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph commented|
|-|-|

As described in the comments on [#7548](https://github.com/twisted/twisted/issues/7548) and the mailing list, we're following a slightly unusual workflow for the tickets in this milestone.  Everything should be reviewed and merged into log-booyah-6750-5 since that code has been reviewed already and is not likely to benefit from another round-trip through the code-review process.

So you want to be looking at [the diff from the branch this should be merged into](https://github.com/twisted/twisted/compare/log-booyah-6750-5...log-remove-legacylogger-7639), not trunk.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_14"></a>|@adiroiban commented|
|-|-|

Ok. I did not see any reference to [#7548](https://github.com/twisted/twisted/issues/7548) in this review request.

It looks good, but there is no news file and no test build for it.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_15"></a>|@wsanchez commented|
|-|-|

adiroiban: Is "looks good" a review modulo the test and new file comments?

Test build is https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/log-remove-legacylogger-7639.

News file will be added when the logging branch is ready to go in, for the overall logging feature.

So if you've reviewed, the code, I think we're OK to merge to the logging branch.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_16"></a>|@wsanchez set status to `closed`|
|-|-|

(In [43570]) Merge branches/log-remove-legacylogger-7639.
Fixes [#7639](https://github.com/twisted/twisted/issues/7639)
